beets/test/helper.py

```diff
@@ -490,99 +456,164 @@ def touch(self, path, dir=None, content=""):
         return path
 
 
+# A test harness for all beets tests.
+# Provides temporary, isolated configuration.
+class BeetsTestCase(unittest.TestCase, TestHelper):
+    """A unittest.TestCase subclass that saves and restores beets'
+    global configuration. This allows tests to make temporary
+    modifications that will then be automatically removed when the test
+    completes. Also provides some additional assertion methods, a
+    temporary directory, and a DummyIO.
+    """
+
+    def setUp(self):
+        self.setup_beets()
+
+    def tearDown(self):
+        self.teardown_beets()
+
+
+class ItemInDBTestCase(BeetsTestCase):
+    """A test case that includes an in-memory library object (`lib`) and
+    an item added to the library (`i`).
+    """
+
+    def setUp(self):
+        super().setUp()
+        self.i = _common.item(self.lib)
+
+
+class PluginMixin:
+    plugin: ClassVar[str]
+
+    def setUp(self):
+        super().setUp()
+        self.load_plugins()
+
+    def tearDown(self):
+        super().tearDown()
+        self.unload_plugins()
+
+
+class PluginTestCase(PluginMixin, BeetsTestCase):
+    pass
+
+
 class ImportHelper(TestHelper):
     """Provides tools to setup a library, a directory containing files that are
     to be imported and an import session. The class also provides stubs for the
     autotagging library and several assertions for the library.
     """
 
-    def setup_beets(self, disk=False):
-        super().setup_beets(disk)
+    resource_path = syspath(os.path.join(_common.RSRC, b"full.mp3"))
+    default_import_config = {
+        "autotag": True,
+        "copy": True,
+        "hardlink": False,
+        "link": False,
+        "move": False,
+        "resume": False,
+        "singletons": False,
+        "timid": True,
+    }
+
+    lib: Library
+    importer: ImportSession
+
+    @cached_property
+    def import_path(self) -> Path:
+        import_path = Path(os.fsdecode(self.temp_dir)) / "import"
+        import_path.mkdir(exist_ok=True)
+        return import_path
+
+    @cached_property
+    def import_dir(self) -> bytes:
+        return bytestring_path(self.import_path)
+
+    def setUp(self):
+        super().setUp()
+        self.import_media = []
         self.lib.path_formats = [
             ("default", os.path.join("$artist", "$album", "$title")),
             ("singleton:true", os.path.join("singletons", "$title")),
             ("comp:true", os.path.join("compilations", "$album", "$title")),
         ]
 
-    def _create_import_dir(self, count=3):
-        """Creates a directory with media files to import.
-        Sets ``self.import_dir`` to the path of the directory. Also sets
-        ``self.import_media`` to a list :class:`MediaFile` for all the files in
-        the directory.
+    def prepare_track_for_import(
+        self,
+        track_id: int,
+        album_path: Path,
+        album_id: int | None = None,
```

Comment:
I had a big comment written out but then I looked it up and PEP 604 was backported through the futures. I'll have to update my typing habits. Great that you're using the more modern way, we'll have to make the change through the codebase (or at least the parts with typing).